### PR TITLE
fix: use gh token if we have one

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry
* [x] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

cc @isuruf

### CODE REVIEW

File: conda_smithy/lint_recipe.py [hunk 1]
```diff
@@ -372,10 +374,21 @@ def lintify_meta_yaml(
 
 def _maintainer_exists(maintainer: str) -> bool:
     """Check if a maintainer exists on GitHub."""
-    return (
-        requests.get(f"https://api.github.com/users/{maintainer}").status_code
-        == 200
-    )
+    if "GH_TOKEN" in os.environ:
+        # use a token if we have one
+        gh = github.Github(auth=github.Auth.Token(os.getenv("GH_TOKEN")))
```
Comment (isuruf) [inline conda_smithy/lint_recipe.py:379]: Doing this every time doubles the number of requests. Put this into a cached function?

**Comment by beckermr** [comment]
shall we merge @isuruf?

File: conda_smithy/lint_recipe.py [hunk 2]
```diff
@@ -370,12 +373,35 @@ def lintify_meta_yaml(
...
 
+# the two functions here allow the cache to refresh
+# if some changes the value of os.environ["GH_TOKEN"]
+# in the same Python process
+@lru_cache(maxsize=1)
+def _cached_gh_with_token(token: str) -> github.Github:
+    return github.Github(auth=github.Auth.Token(token))
+
+
+def _cached_gh():
```
Comment (beckermr) [inline conda_smithy/lint_recipe.py:384]:
```suggestion
def _cached_gh(): -> github.Github:
```

**Comment by beckermr** [comment]
pre-commit.ci autofix

File: conda_smithy/lint_recipe.py [hunk 3]
```diff
@@ -370,12 +373,35 @@ def lintify_meta_yaml(
...
 
+# the two functions here allow the cache to refresh
+# if some changes the value of os.environ["GH_TOKEN"]
+# in the same Python process
+@lru_cache(maxsize=1)
+def _cached_gh_with_token(token: str) -> github.Github:
+    return github.Github(auth=github.Auth.Token(token))
+
+
+def _cached_gh(): -> github.Github:
```
Comment (beckermr) [inline conda_smithy/lint_recipe.py:384]:
```suggestion
def _cached_gh() -> github.Github:
```

**Review by isuruf — APPROVED** [review]

**Pull request merged**